app/jobs/embargo_expiry_job.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+class EmbargoExpiryJob < Hyrax::ApplicationJob
+  def perform
+    records_with_expired_embargos.each do |record|
+      ExpireEmbargoJob.perform_later(record)
+    end
+  end
+
+  def records_with_expired_embargos
```

Comment:
I wonder if this is really necessary and if the presenters returned by `Hyrax::EmbargoService` can be passed directly into `perform_later`.  Could you try that and see if it works?

Reply:
I believe that I did and it did not but just for you Chris I will try again

Reply:
If it doesn't automagically work, then it might be necessary to add a `to_global_id` method to the presenter.  I think that could just be delegated in [ModelProxy](https://github.com/samvera/hyrax/blob/08ffa5aa73ee81b2c812d81635bfa328ca565f53/app/presenters/hyrax/model_proxy.rb#L7).

Reply:
it does seem like we need a better pattern here. the map into `AF::Base.find` on L11 seems like it could be very expensive.

i'm not sure adding `to_global_id` to the presenter is great, though. i think `GlobalId` is really meant to uniquely identify the model, and the presenter is a different model from the `AF::Base` one, right?

a fast, code-smelly fix might be 

```ruby
assets = Hyrax::EmbargoService.assets_with_expired_embargoes
assets.map { |presenter| presenter.solr_document.to_model }
```

there's also a valkyrie powered, [`Hyrax::EmbargoManager`](https://github.com/samvera/hyrax/blob/130c4e600318a9194725b35dd0fd6e19e5108dd9/app/services/hyrax/embargo_manager.rb) which this will eventually (soon) need to be rewritten to use. if you're up for it, it would be worth reworking this now rather than later.

Reply:
@no-reply 

> there's also a valkyrie powered, Hyrax::EmbargoManager which this will eventually (soon) need to be rewritten to use.

When you say this do you mean this method or other parts of this workflow, because I don't see anything in `Hyrax::EmbargoManager` that gives me a list of all assets with expired embargoes, but I can see how we could use it to rewrite `ExpireEmbargoJob` for example

Reply:
yes. sorry if my comment was misleading. the `EmbargoManager` handles the embargo workflow for individual objects. i think the `Hyrax::EmbargoService` is still the best way to query embargo status in bulk.